test/SparqlLexerTest.cpp

```diff
@@ -182,3 +182,14 @@ TEST(SparqlLexer, reset) {
   lexer.expect("<cs.uni-freiburg.de/>");
   ASSERT_TRUE(lexer.getUnconsumedInput().empty());
 }
+
+TEST(SparqlLexer, peek) {
+  std::string query = "PREFIX";
+
+  SparqlLexer lexer(query);
+  ASSERT_TRUE(lexer.peek("prefix"));
+  ASSERT_EQ(std::string("prefix"), lexer.getUnconsumedInput());
+
+  ASSERT_TRUE(lexer.peek(SparqlToken::Type::KEYWORD));
+  ASSERT_EQ(std::string("prefix"), lexer.getUnconsumedInput());
```

Comment:
Please also add a test case where the peek fails, and for the case-insensitivity.

Reply:
I'll add a failing test.

The problem with `match_case` is that the lexer makes everything lowercase anyways. A test that tries to match a non lowercase string with `match_case` can of course still be constructed. But I don't really see any added value there at the moment for that reason.